pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -22,18 +22,26 @@ func handleSyncOp(ctx *downloaderContext, key string,
 	config types.DownloaderConfig, status *types.DownloaderStatus,
 	dst *types.DatastoreConfig) {
 	var (
-		err                                        error
-		errStr, locFilename, remoteName, serverURL string
-		syncOp                                     zedUpload.SyncOpType = zedUpload.SyncOpDownload
-		trType                                     zedUpload.SyncTransportType
-		auth                                       *zedUpload.AuthInput
+		err                                                              error
+		errStr, locFilename, locDirname, filename, remoteName, serverURL string
+		syncOp                                                           zedUpload.SyncOpType = zedUpload.SyncOpDownload
+		trType                                                           zedUpload.SyncTransportType
+		auth                                                             *zedUpload.AuthInput
 	)
 
 	if status.ObjType == "" {
 		log.Fatalf("handleSyncOp: No ObjType for %s",
 			status.ImageID)
 	}
 
+	// the target filename, where to place the download, is provided in config.
+	// downloader is welcome to use any other local caching directories it wants,
+	// at its sole discretion, but config.Target is where it is expected to place
```

Comment:
It downloader allows to use other files in locDirname for caching? Or does it need go use something completely different?
That affects whether this code should clean up the whole directory on error, or just delete any locFilename=Target on error.

Reply:
No, it cannot use any other files in `locDirname`, but it can use its _own_ local caching.

What I was trying to capture was the following patterns. 

1. downloader receives a request to download a file to `target` location
1. downloader can do one of the following, at its sole discretion:
   * download the content to `target` and report back when it is full
   * create some local caching directory that it controls, download part or all to there, and place the results in `target` when it is ready in a single atomic activity

If it does the second option, then it is up to downloader to clean up the cache.

I will update the comment.